src/profile/components/multiItemAddressRow/MultiItemAddressRow.tsx

```diff
@@ -57,7 +57,8 @@ function MultiItemAddressRow(props: RowItemProps): React.ReactElement {
     FormikValues
   >(t, isNew);
 
-  const { primary } = data;
+  const { primary, saving } = data;
+  const disableButtons = !!currentAction || !!saving;
 
```

Comment:
The "currentAction" is not the only indicator for item being saved. "saving" property can also change when item is primary and another item is being set as primary.